compiler/rustc_hir/src/hir.rs

```diff
@@ -2275,21 +2275,15 @@ pub enum ImplItemKind<'hir> {
     Type(&'hir Ty<'hir>),
 }
 
-/// Bind a type to an associated type (i.e., `A = Foo`).
+/// An associated item binding.
 ///
-/// Bindings like `A: Debug` are represented as a special type `A =
-/// $::Debug` that is understood by the astconv code.
+/// ### Examples
 ///
-/// FIXME(alexreg): why have a separate type for the binding case,
```

Comment:
I never understood this FIXME whenever I came across these docs: The FIXME was created in the same commit as the one that introduced `TypeBindingKind` in favor of the now gone struct field `ty: P<Ty>` the FIXME mentions...